fabric-rendering-fluids-v1/src/main/java/net/fabricmc/fabric/impl/client/rendering/fluid/FluidRenderHandlerRegistryImpl.java

```diff
@@ -64,6 +66,17 @@ public FluidRenderHandler getOverride(Fluid fluid) {
 	public void register(Fluid fluid, FluidRenderHandler renderer) {
 		handlers.put(fluid, renderer);
 		modHandlers.put(fluid, renderer);
+
+		Identifier[] textures = renderer.getTexturesIds();
+
+		if (textures.length > 0) {
+			ClientSpriteRegistryCallback.event(PlayerScreenHandler.BLOCK_ATLAS_TEXTURE).register((atlasTexture, registry) -> {
```

Comment:
Registing this was left to the modder before, I think this now has the possibility to start registering textures that werent before. Possibly leading to errors in the log? Please correct me if im wrong. 👍

Reply:
Fixed

Reply:
I've changed the behaviour by restoring the default behaviour and adding specific constructors to SimpleFluidRenderHandler by using a RegistrableTexture object to specify if to register or not the textures.
Please check this ;)

https://github.com/FabricMC/fabric/blob/01c4d2d5a7a3ac890e1647adbd46083553f02104/fabric-rendering-fluids-v1/src/main/java/net/fabricmc/fabric/api/client/render/fluid/v1/SimpleFluidRenderHandler.java

https://github.com/FabricMC/fabric/blob/01c4d2d5a7a3ac890e1647adbd46083553f02104/fabric-textures-v0/src/main/java/net/fabricmc/fabric/api/client/texture/RegistrableTexture.java